### PR TITLE
chore: #1634 added breaking changes label to release notes

### PR DESCRIPTION
**Pull request by wli-pro**
### Context

Implemeneted #1634 into release notes

### Labels

Please apply following PR-related labels when appropriate:
- `C0-breaking`: if your change could break the existing client, e.g. API change, critical logic change 
- `C1-noteworthy`: if your change is non-breaking, but is still worth noticing for the client, e.g. reference code improvement

### Testing Evidences

Please attach any relevent evidences if applicable

https://github.com/litentry/litentry-parachain/assets/120463031/b89955ae-14aa-4953-b4df-d0cd69beddc6

### CODE REVIEW

File: .github/workflows/create-release-draft.yml [hunk 1]
```diff
@@ -278,6 +278,8 @@ jobs:
       - name: Generate release notes
         run: |
           ./scripts/generate-release-notes.sh ${{ github.workspace }}/.github/release_notes.md ${{ needs.set-release-type.outputs.release_type }} ${{ env.DIFF_TAG }}
+        env:
+          GH_TOKEN: ${{secrets.GITHUB_TOKEN}}
```
Comment (Kailai-Wang) [inline .github/workflows/create-release-draft.yml:282]: Is this required?

File: scripts/generate-release-notes.sh [hunk 2]
```diff
@@ -161,16 +161,24 @@ else
...
 Details:
 
 EOF
 
-  # use %n as a workaround otherwise there's no newline after the whole body
-  git log --no-merges --abbrev-commit --pretty="format:%h|%s%n" $DIFF_TAG..$RELEASE_TAG | grep -v "^$" | while read -r f; do
-    commit=$(echo "$f" | cut -d'|' -f1)
-    desc=$(echo "$f" | cut -d'|' -f2)
-    echo -e "- [\`$commit\`]($REPO/commit/$commit) $desc" >> "$1"
+labels=("C0-breaking" "C1-noteworthy")
```
Comment (Kailai-Wang) [inline scripts/generate-release-notes.sh]: Maybe the indention :)

**Review by Kailai-Wang — APPROVED** [review]
Looks good, thanks, only two small comments.

File: .github/workflows/create-release-draft.yml (hunk 1 above)
Comment (wli-pro) [reply .github/workflows/create-release-draft.yml:282]: gh pr list should need it.

Comment (Kailai-Wang) [reply .github/workflows/create-release-draft.yml:282]: Maybe not - https://cli.github.com/manual/gh_help_environment
`GITHUB_TOKEN` is present by default, can you please find it out if we need this extra env?

Comment (wli-pro) [reply .github/workflows/create-release-draft.yml:282]: tried below code:
image


Result:

image


So looks like GH_TOKEN or GITHUB_TOKEN is not in the env

File: scripts/generate-release-notes.sh [hunk 3]
```diff
@@ -161,16 +161,24 @@ else
...
 ## Changes
 
-Raw diff: [$DIFF_TAG...$RELEASE_TAG](https://github.com/litentry/litentry-parachain/compare/$DIFF_TAG...$RELEASE_TAG)
+Raw diff: [$DIFF_TAG...$RELEASE_TAG]($REPO/compare/$DIFF_TAG...$RELEASE_TAG)
 
 Details:
 
 EOF
 
-  # use %n as a workaround otherwise there's no newline after the whole body
```
Comment (BillyWooo) [inline scripts/generate-release-notes.sh:170]: why rm this comment line?

File: .github/workflows/create-release-draft.yml (hunk 1 above)
Comment (BillyWooo) [reply .github/workflows/create-release-draft.yml:282]: Yes. There is `secrets.GITHUB_TOKEN`, but not `env.GH_TOKEN`. Where do we need `env.GH_TOKEN`? If so, why not use `secrets.GITHUB_TOKEN` directly?

File: scripts/generate-release-notes.sh [hunk 4]
```diff
@@ -161,16 +161,24 @@ else
...
+
+  git log --no-merges --abbrev-commit --pretty="format:%h|%s%n" "$DIFF_TAG..$RELEASE_TAG" | grep -v "^$" | while read -r f; do
     commit=$(echo "$f" | cut -d'|' -f1)
     desc=$(echo "$f" | cut -d'|' -f2)
-    echo -e "- [\`$commit\`]($REPO/commit/$commit) $desc" >> "$1"
+    output="- [\`$commit\`]($REPO/commit/$commit) $desc"
+    
+    for ((i=0; i<${#labels[@]}; i++)); do
+      label=$(gh pr list --search "$commit" --label "${labels[i]}" --state merged)
+      [ -n "$label" ] && output+=" $REPO/labels/${labels[i]}"
```
Comment (BillyWooo) [inline scripts/generate-release-notes.sh:179]: Couldn't get a chance to try it. But would it be better to add some new line at the end of each line?

Comment (wli-pro) [reply scripts/generate-release-notes.sh:179]: The testing evidence in the description of this PR already shows the visual looks ok with line break already.

File: scripts/generate-release-notes.sh (hunk 3 above)
Comment (wli-pro) [reply scripts/generate-release-notes.sh:170]: well spot and added it back now.

File: .github/workflows/create-release-draft.yml (hunk 1 above)
Comment (wli-pro) [reply .github/workflows/create-release-draft.yml:282]: gh pr list --search needs it

This is the suggested way from github, see the button part of the image: 

https://user-images.githubusercontent.com/120463031/250709186-69a435a0-b167-4a1f-b1e8-d3ad70afcbfb.png

**Pull request merged**